server/src/pages/shot/editor.js

```diff
@@ -61,10 +95,239 @@ exports.Editor = class Editor extends React.Component {
+      let moveY = movement[1];
+      if (moveY.includes("*")) {
+        selectedPos.top = Math.max(0, movingPos.top + diffY);
+        selectedPos.bottom = Math.min(this.canvasHeight, movingPos.bottom + diffY);
+      }
+      if (moveY.includes('y2')) {
+        selectedPos.bottom = Math.min(this.canvasHeight, resizeStartPos.y + diffY);
+      }
+      if (moveY.includes('y1')) {
+        selectedPos.top = Math.max(0, resizeStartPos.y + diffY);
+      }
+    }
+    selectedPos = this.flipCoords(selectedPos);
+    this.displayCropBox(selectedPos, direction);
+  }
+
+  flipCoords(pos) {
+    let temp;
+    if (pos.right <= pos.left) {
+      temp = pos.left;
+      pos.left = pos.right;
+      pos.right = temp;
+    }
+    if (pos.bottom <= pos.top) {
+      temp = pos.top;
+      pos.top = pos.bottom;
+      pos.bottom = temp;
+    }
+    return pos;
+  }
+
+  findClickedArea(target) {
+    let movements = ["topLeft", "top", "topRight", "left", "right", "bottomLeft", "bottom", "bottomRight"];
+      if (target.classList.contains("mover-target") || target.classList.contains("mover")) {
+        for (let name of movements) {
+          if (target.classList.contains("direction-" + name) || target.parentNode.classList.contains("direction-" + name)) {
+            return name;
+          }
+        }
+      } else if (target.classList.contains("highlight")) {
+        return "move";
+      }
+    if (target.classList.contains("bghighlight")) {
+      this.onClickCancelCrop();
+    }
+    return null;
+  }
+
+  removeCropBox() {
+    while (this.cropContainer.firstChild) {
+      this.cropContainer.firstChild.remove();
+    }
+    this.cropBox = null;
+    selectedPos = {};
+    movingPos = {};
+    resizeDirection = null;
+    resizing = false;
+    creating = false;
+  }
+
+  displayCropBox(pos, direction) {
+    this.createCropBox();
+    this.cropBox.style.position = "absolute";
+    this.cropBox.style.top = pos.top + "px";
+    this.cropBox.style.left = pos.left + "px";
+    this.cropBox.style.height = pos.bottom - pos.top + "px";
+    this.cropBox.style.width = pos.right - pos.left + "px";
+    this.bgTop.style.top = "0px";
+    this.bgTop.style.height = pos.top + "px";
+    this.bgTop.style.left = "0px";
+    this.bgTop.style.width = "100%";
+    this.bgBottom.style.top = pos.bottom + "px";
+    this.bgBottom.style.height = "100%";
+    this.bgBottom.style.left = "0px";
+    this.bgBottom.style.width = "100%";
+    this.bgLeft.style.top = pos.top + "px";
+    this.bgLeft.style.height = pos.bottom - pos.top + "px";
+    this.bgLeft.style.left = "0px";
+    this.bgLeft.style.width = pos.left + "px";
+    this.bgRight.style.top = pos.top + "px";
+    this.bgRight.style.height = pos.bottom - pos.top + "px";
+    this.bgRight.style.left = pos.right + "px";
+    this.bgRight.style.width = "100%";
+  }
+
+  createCropBox() {
+    let movements = ["topLeft", "top", "topRight", "left", "right", "bottomLeft", "bottom", "bottomRight"];
```

Comment:
This line is identical to the one in `findClickedArea`, and both have the same name as another variable in the outer scope.  Perhaps rename it and move it to the outer scope.